### PR TITLE
Add YAML output  in cluster dump and analyze and remove managed fields by default 

### PR DESCRIPTION
**Pull request by George-Aeillo**
In this PR, I add support for YAML in the `cluster dump` and `analyze`, making it the default behavior, unless the -j is specified. With cluster `info`, a YAML dump is always used, it is also used when analyzing a live cluster. A bug was also fixed, where the `cluster analyze` command would not error if an incorrect path was given. The managed fields portion of Kubernetes resources was also removed by default. 

Example of output of the namespace resource list in default/YAML format with no managed Fields

```
./ocne cluster dump  -d  test-yaml-dump
INFO[2024-11-12T16:49:47-05:00] Collecting node data                         
INFO[2024-11-12T16:49:47-05:00] Collecting cluster data                      
INFO[2024-11-12T16:49:57-05:00] Cluster dump successfully completed 

cat /cluster/cluster-wide/namespaces.yaml
apiVersion: v1
items:
- apiVersion: v1
  kind: Namespace
  metadata:
    creationTimestamp: "2024-11-07T20:03:42Z"
    labels:
      kubernetes.io/metadata.name: default
    name: default
    resourceVersion: "16"
    uid: 86875dcb-0064-4161-bb64-2553960c9deb
  spec:
    finalizers:
    - kubernetes
  status:
    phase: Active
...
kind: NamespaceList
metadata:
  resourceVersion: "121952"

```
Example of analysis run on a live cluster
```
./ocne cluster analyze   
INFO[2024-11-12T16:47:41-05:00] Collecting node data                         

Cluster Nodes:
--------------
Cluster nodes are normal

Pods:
-----
Pods are normal

```

Example of output of the namespace resource list in JSON with format and managed Fields
```
./ocne cluster dump --json --managed -d ./test-json-dump
INFO[2024-11-12T16:49:47-05:00] Collecting node data                         
INFO[2024-11-12T16:49:47-05:00] Collecting cluster data                      
INFO[2024-11-12T16:49:57-05:00] Cluster dump successfully completed 
cat /cluster/cluster-wide/namespaces.json

    {
...
      "apiVersion": "v1",
      "kind": "Namespace",
      "metadata": {
        "creationTimestamp": "2024-11-07T20:03:42Z",
        "labels": {
          "kubernetes.io/metadata.name": "kube-node-lease"
        },
        "managedFields": [
          {
            "apiVersion": "v1",
            "fieldsType": "FieldsV1",
            "fieldsV1": {
              "f:metadata": {
                "f:labels": {
                  ".": {},
                  "f:kubernetes.io/metadata.name": {}
                }
              }
            },
            "manager": "kube-apiserver",
            "operation": "Update",
            "time": "2024-11-07T20:03:42Z"
          }
        ],
        "name": "kube-node-lease",
        "resourceVersion": "14",
        "uid": "e79acdfb-e92c-4abc-9c47-4dc09400468f"
      },
      "spec": {
        "finalizers": [
          "kubernetes"
        ]
      },
      "status": {
        "phase": "Active"
      }
    },
...
```

Example of analysis run on a cluster dump that has JSON and managed fields 
```
./ocne cluster analyze --json  -d ./test-json-dump                     

Cluster Nodes:
--------------
Cluster nodes are normal

Pods:
-----
Pods are normal

```

Example of analysis run on a cluster dump that has YAML and no managed fields (default output) 
```
./ocne cluster analyze -d  ./test-yaml-dump 

Cluster Nodes:
--------------
Cluster nodes are normal

Pods:
-----
Pods are normal

```

Example of analysis run on a cluster dump path that does not exist 
```
./ocne cluster analyze -d  ./test-does-not-exist
ERRO[2024-11-12T17:00:22-05:00] stat ./test-does-not-exist: no such file or directory 

```

### CODE REVIEW

File: cmd/cluster/dump/dump.go [hunk 1]
```diff
@@ -82,6 +82,14 @@ const (
 	flagGenerateArchive      = "generate-archive"
 	flagGenerateArchiveShort = "z"
 	flagGenerateClusterHelp  = "Generate an archive instead of dumping files to an output directory.  The filename must end with .tgz or .tar.gz"
+
+	flagManaged      = "managed"
+	flagManagedShort = "f"
```
Comment (pfmackin) [inline cmd/cluster/dump/dump.go:87]: can you change "f" to "g"

**Review by pfmackin — DISMISSED** [review]

File: cmd/cluster/analyze/analyze.go [hunk 2]
```diff
@@ -40,6 +40,10 @@ const (
 	flagVerbose      = "verbose"
 	flagVerboseShort = "v"
 	flagVerboseHelp  = "Display additional detailed information related to the analysis."
+
+	flagJSON      = "json"
+	flagJSONShort = "j"
```
Comment (mgianatagh) [inline cmd/cluster/analyze/analyze.go:45]: Can we do without this shortcut?  I am not sure we need to have one for every option.

File: cmd/cluster/dump/dump.go (hunk 1 above)
Comment (George-Aeillo) [reply cmd/cluster/dump/dump.go]: Ok! I removed the short.

File: cmd/cluster/analyze/analyze.go (hunk 2 above)
Comment (George-Aeillo) [reply cmd/cluster/analyze/analyze.go]: Ok! I removed the shortcut flag for this one.

**Review by pfmackin — APPROVED** [review]

**Pull request merged**